File: src/Core/Dating.Application.Contracts/Commands/UpdateProfileCommand.cs
namespace Dating.Application.Contracts.Commands;

public class UpdateProfileCommand : RequestBase
{
    public UpdateProfileCommand(string id)
    {
        Id = id;
    }
    
    public string Id { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? LivingCity { get; set; }
    public string? School { get; set; }
    public string? JobTitle { get; set; }
    public string? Company { get; set; }
    public string? Bio { get; set; }
    public Gender? Gender { get; set; }
    public SexualOrientation? Orientation { get; set; }
    public DateTime? Birthdate { get; set; }
}

File: src/Client/Dating.Web/src/app/shared/models/commands/updateProfileCommand.ts
import {Gender, SexualOrientation} from '../../types';

export interface UpdateProfileCommand {
  id: string;
  firstName?: string;
  lastName?: string;
  birthdate?: string;
  livingCity?: string;
  gender?: Gender;
  orientation?: SexualOrientation;
  school?: string;
  jobTitle?: string;
  company?: string;
  bio?: string;
}
